python/labbox_ephys/extractors/_in_memory.py

```diff
@@ -0,0 +1,24 @@
+import random
+
+
+def _random_string(num_chars: int) -> str:
+    chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
+    return ''.join(random.choice(chars) for _ in range(num_chars))
+
+unique_process_id = _random_string(20)
+in_memory_objects = {}
+
+def register_in_memory_object(obj):
```

Comment:
This is a way to register and retrieve in-memory objects. The dict() that is returned will be unique to that object for this python process.